client/v2/indexer/lookupAccountTransactions.go

```diff
@@ -0,0 +1,100 @@
+package indexer
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+)
+
+type LookupAccountTransactions struct {
+	c       *Client
+	account string
+	p       models.LookupAccountTransactionsParams
+}
+
+func (s *LookupAccountTransactions) NextToken(nextToken string) *LookupAccountTransactions {
+	s.p.NextToken = nextToken
+	return s
+}
+
+func (s *LookupAccountTransactions) NotePrefix(prefix []byte) *LookupAccountTransactions {
+	s.p.NotePrefix = base64.StdEncoding.EncodeToString(prefix)
+	return s
+}
+
+func (s *LookupAccountTransactions) TxType(txtype string) *LookupAccountTransactions {
+	s.p.TxType = txtype
+	return s
+}
+
+func (s *LookupAccountTransactions) SigType(sigtype string) *LookupAccountTransactions {
+	s.p.SigType = sigtype
+	return s
+}
+
+func (s *LookupAccountTransactions) TXID(txid string) *LookupAccountTransactions {
+	s.p.TxId = txid
+	return s
+}
+
+func (s *LookupAccountTransactions) Round(rnd uint64) *LookupAccountTransactions {
+	s.p.Round = rnd
+	return s
+}
+
+func (s *LookupAccountTransactions) MinRound(min uint64) *LookupAccountTransactions {
+	s.p.MinRound = min
+	return s
+}
+
+func (s *LookupAccountTransactions) MaxRound(max uint64) *LookupAccountTransactions {
+	s.p.MaxRound = max
+	return s
+}
+
+func (s *LookupAccountTransactions) AssetID(index uint64) *LookupAccountTransactions {
+	s.p.AssetId = index
+	return s
+}
+
+func (s *LookupAccountTransactions) Limit(limit uint64) *LookupAccountTransactions {
+	s.p.Limit = limit
+	return s
+}
+
+func (s *LookupAccountTransactions) BeforeTime(before string) *LookupAccountTransactions {
```

Comment:
nit: would be nice to have a date type here, building RFC3339 strings can be annoying.